### ISSUE TITLE
Unmanaged generic constraint is causing invalid code generation

### ISSUE THREAD

**Issue by BasicallyIAmFox** [mattwhitfield/Unitverse#248]
**Describe the bug**
Using `unmanaged` generic constraint produces non-compiling code.

**To Reproduce**
Generating tests for following type:
```cs
public struct Foo<T> where T : unmanaged {
  public void Bar() { }
}
```
Produces:
```cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using T = System.String;

[TestClass]
public class Foo_1Tests {
	private Foo<T> _testClass;

	[TestInitialize]
	public void SetUp() {
		_testClass = new Foo<T>();
	}

	[TestMethod]
	public void CanCallBar() {
		// Act
		_testClass.Bar();

		// Assert
		Assert.Fail("Create or modify test");
	}
}
```

**Comment by mattwhitfield** [comment]
Thanks for the report. I'm interested as to what you would want it to do here. There are a bunch on type constraints that Unitverse doesn't support. I did previously think about something that could generate a type that satisfied all the constraints - but I thought that was a lot of effort to generate something that would likely be thrown away.

So I am interested to hear your thoughts...

**Comment by BasicallyIAmFox** [comment]
> Thanks for the report. I'm interested as to what you would want it to do here. There are a bunch on type constraints that Unitverse doesn't support. I did previously think about something that could generate a type that satisfied all the constraints - but I thought that was a lot of effort to generate something that would likely be thrown away.
> 
> So I am interested to hear your thoughts...

primitive types are unmanaged, so they would work for sure

**Comment by mattwhitfield** [comment]
This is resolved in 0.208 - it effectively replaces `using T = System.String` with `using T = System.Int32` if the `struct` or `unmanaged` constraints are present. It doesn't try and fully solve the constraint problem.
